client/assignments.go

```diff
@@ -38,10 +38,10 @@ type HTTPTestHeader struct {
 	Value string
 }
 
-type Assignment struct {
-	Assignment struct {
-		Type                    string
-		AssignmentDataHTTPTests *struct {
+type Lesson struct {
```

Comment:
this is what has to change if we do the breaking change of changing assignments to lessons, sorry.